phoenix-core/src/main/java/org/apache/phoenix/schema/PMetaDataImpl.java

```diff
@@ -160,7 +160,7 @@ public PMetaDataCache cloneMinusOverage(long overage) {
                     newCache.put(tableRef.table.getKey(), new PTableRef(tableRef));
                     toRemove.add(tableRef);
                     toRemoveBytes += tableRef.estSize;
-                    if (toRemoveBytes - toRemove.peekLast().estSize > overage) {
+                    while (toRemoveBytes - toRemove.peekLast().estSize >= overage) {
```

Comment:
I did think about that but I don't think it's possible for it to be empty. If you hit the line for the first time, it's just after you've added something; if you hit it on a subsequent iteration, it's because you decided to remove something that left `toRemoveBytes` strictly positive, which means there must still be something left in the eviction queue.